troubleshoot/elasticsearch/high-cpu-usage.md

```diff
@@ -77,17 +75,54 @@ This API returns a breakdown of any hot threads in plain text. High CPU usage fr
 
 The following tips outline the most common causes of high CPU usage and their solutions.
 
-**Scale your cluster**
+**Check JVM garbage collection**
+
+High CPU usage is often caused by excessive JVM garbage collection (GC) activity. This excessive GC typically arises from configuration problems or inefficient queries causing increased heap memory usage.
+
+For optimal JVM performance, garbage collection should meet these criteria:
+
+1. Young GC completes quickly (ideally within 50 ms).
+2. Young GC does not occur too frequently (approximately once every 10 seconds).
+3. Old GC completes quickly (ideally within 1 second).
+4. Old GC does not occur too frequently (once every 10 minutes or less frequently).
+
+Excessive JVM garbage collection usually indicates high heap memory usage. Common potential reasons for increased heap memory usage include:
+
+* Oversharding of indices
+* Very large aggregation queries
+* Excessively large bulk indexing requests
+* Inefficient or incorrect mapping definitions
+* Improper heap size configuration
+* Misconfiguration of JVM new generation ratio (-XX:NewRatio)
+
+**Hot spotting**
+
+You might experience high CPU usage on specific data nodes or an entire [data tier](/manage-data/lifecycle/data-tiers.md) if traffic isn’t evenly distributed—a scenario known as [hot spotting](hotspotting.md). This commonly occurs when read or write applications don’t properly balance requests across nodes, or when indices receiving heavy write activity (like hot-tier indices) have their shards concentrated on just one or a few nodes.
+
+For details on diagnosing and resolving these issues, see [hot spotting](hotspotting.md).
+
+**Oversharding**
+
+If your Elasticsearch cluster contains a large number of shards, you might be facing an oversharding issue.
+
+Oversharding occurs when there are too many shards, causing each shard to be smaller than optimal. While Elasticsearch doesn’t have a strict minimum shard size, an excessive number of small shards can negatively impact performance. Each shard consumes cluster resources since Elasticsearch must maintain metadata and manage shard states across all nodes.
```

Comment:
```suggestion
Oversharding occurs when there are too many shards, causing each shard to be smaller than optimal. While {{es}} doesn’t have a strict minimum shard size, an excessive number of small shards can negatively impact performance. Each shard consumes cluster resources, because {{es}} must maintain metadata and manage shard states across all nodes.
```